Scenes/Levels/TemplateLevel/Navigation.gd

```diff
@@ -1,16 +1,123 @@
 extends TileMap
 
-const TOWER_SIZE = 64
-const BUFFER = 1
+# Mostly copied from https://escada-games.itch.io/randungeon/devlog/261991/how-to-use-godots-astar2d-for-path-finding
+
+var aStar:AStar2D
+export var debug_mode = false setget set_debug_mode
+export (NodePath) var goal
+onready var goal_pos = get_node(goal).global_position
+const GOAL_IDX = 69420
+
+var target_texture = preload("res://Assets/GFX/pointers/crosshair063.png")
+
 func _ready():
-	update_nav_area()
-
-func update_nav_area():
-	for tower in $"../Towers".get_children():
-		var starting_cell_v = tower.global_position/cell_size
-		for offset_x in range(-BUFFER, TOWER_SIZE/cell_size.x + BUFFER):
-			for offset_y in range(-BUFFER, TOWER_SIZE/cell_size.y + BUFFER):
-				var offset_v = Vector2(offset_x, offset_y)
-				set_cellv(starting_cell_v + offset_v, -1)
-	update_dirty_quadrants()
-	get_tree().call_group("Navigators", "update_path")
+	aStar = AStar2D.new()
+	for cell in get_used_cells():
+		var idx = get_AStar_cell_id(cell)
+		var point = get_cell_point(cell)
+		aStar.add_point(idx, point)
+	for cell in get_used_cells():
+		for neighbor in get_neighboring_cells(cell):
+			aStar.connect_points(get_AStar_cell_id(cell), get_AStar_cell_id(neighbor))
+	aStar.add_point(GOAL_IDX, goal_pos)
+	aStar.connect_points(GOAL_IDX, get_AStar_cell_id(world_to_map(goal_pos)))
+	if debug_mode:
+		draw_debug_points()
+
+func get_path_to_goal(from: Vector2):
```

Comment:
is it necessary to specify the input type? since we're not doing it anywhere else, we could remove it from this file to keep it consistent

Reply:
I did it like this because it gets called by other nodes so it helps to have the inellisense on the parameter to reinforce that it takes coordinates

Reply:
That being said I would prefer to add types to more places rather than go the other way

Reply:
The only place I have encountered where you can't add types is signals. Also, I don't think there is support for complex types (like an array of Vectors)

Reply:
> That being said I would prefer to add types to more places rather than go the other way

down for this